addons/test/misra/misra-test.c

```diff
@@ -769,6 +769,7 @@ struct misra_10_6_s {
 static void misra_10_6(u8 x, char c1, char c2) {
   u16 y1 = x+x; // 10.6
   u16 y2 = (0x100u - 0x80u); // rhs is not a composite expression because it's a constant expression
+  int b = (y2 == y2) ? 0 : 1; // no-warning
```

Comment:
could you clarify.. is the idea that this is not a composite expression? as the variable y2 is used when calculating the result I would spontanously guess it should be a composite expression?

Reply:
conditional operators are a special case and are only considered a composite operator, if either the second or third operand is a composite expression. So this isn't a composite expression. See section 8.10.3 for reference (not sure if I can copy paste the content from the misra handbook here)

Reply:
ok I can see it.